src/c-writer.cc

```diff
@@ -1790,15 +1802,50 @@ void CWriter::WriteCallIndirectFuncDeclaration(const FuncDeclaration& decl,
   Write(")");
 }
 
-void CWriter::WriteFeatureMacros() {
+static bool func_uses_simd(const FuncSignature& sig) {
+  return std::any_of(sig.param_types.begin(), sig.param_types.end(),
+                     [](auto x) { return x == Type::V128; }) ||
+         std::any_of(sig.result_types.begin(), sig.result_types.end(),
+                     [](auto x) { return x == Type::V128; });
+}
+
+void CWriter::ComputeSimdScope() {
+  simd_used_in_header_ =
+      module_->features_used.simd &&
+      (std::any_of(module_->globals.begin(), module_->globals.end(),
+                   [](const auto& x) { return x->type == Type::V128; }) ||
+       std::any_of(module_->imports.begin(), module_->imports.end(),
+                   [](const auto& x) {
+                     return x->kind() == ExternalKind::Func &&
+                            func_uses_simd(cast<FuncImport>(x)->func.decl.sig);
+                   }) ||
+       std::any_of(module_->exports.begin(), module_->exports.end(),
+                   [&](const auto& x) {
+                     return x->kind == ExternalKind::Func &&
+                            func_uses_simd(module_->GetFunc(x->var)->decl.sig);
+                   }));
```

Comment:
How come we have have all this nice modern style of C++ but we still have to type of out `.begin()` and `.end()` everywhere..  :(

Also, why does the third lambda have the `&` capture specifier?   (sorry I'm still not used to reading C++ lambdas).

Reply:
:-) I think if we upgraded to C++20 we could probably use `std::ranges::any_of`, which finally gets rid of the .begin() and .end().

The third lambda is the only one that uses the `module_` member, so it needs to capture it (and capturing implicitly with `&` seemed like the simplest way to do that).